zanata-client-commands/src/main/java/org/zanata/client/commands/init/TransConfigPrompt.java

```diff
@@ -123,19 +132,61 @@ TransConfigPrompt promptUser() throws Exception {
 
     private static class ToTransFileNameFunction
             implements Function<String, String> {
-        private final PullStrategy strategy;
+        private final TransFilePathFinder transFilePathFinder;
         private final LocaleMapping localeMapping;
 
-        public ToTransFileNameFunction(PullStrategy strategy,
+        public ToTransFileNameFunction(TransFilePathFinder transFilePathFinder,
                 LocaleMapping localeMapping) {
-            this.strategy = strategy;
+            this.transFilePathFinder = transFilePathFinder;
             this.localeMapping = localeMapping;
         }
 
         @Override
         public String apply(String input) {
-            return strategy.getTransFileToWrite(input, localeMapping).getPath();
+            return transFilePathFinder.getTransFileToWrite(input, localeMapping);
+
+        }
+    }
+
+    interface TransFilePathFinder {
+        String getTransFileToWrite(String srcDoc, LocaleMapping localeMapping);
+    }
+
+    static class RawTransFilePathFinder implements TransFilePathFinder {
+        private final TransFileResolver transFileResolver;
+
+        RawTransFilePathFinder(PullOptions opts) {
+            transFileResolver = new TransFileResolver(opts);
+        }
 
+        @Override
+        public String getTransFileToWrite(String srcDoc,
+                LocaleMapping localeMapping) {
+            String targetFileExt = FilenameUtils
+                    .getExtension(srcDoc);
+
+            Optional<String> translationFileExtension =
+                    Optional.fromNullable(targetFileExt);
+            File file = transFileResolver.resolveTransFile(
+                    QualifiedSrcDocName.from(srcDoc),
+                    localeMapping, translationFileExtension);
+            return file.getPath();
+        }
+    }
+
+    static class OtherTransFilePathFinder implements TransFilePathFinder {
```

Comment:
If it is specifically not raw files, I suggest `NonRawTransFilePathFinder`, since "Other" is not meaningful on its own.